docs/source/onedal/algorithms/graph/subgraph-isomorphism.rst

```diff
@@ -0,0 +1,94 @@
+.. * Copyright 2021 Intel Corporation
+.. *
+.. * Licensed under the Apache License, Version 2.0 (the "License");
+.. * you may not use this file except in compliance with the License.
+.. * You may obtain a copy of the License at
+.. *
+.. *     http://www.apache.org/licenses/LICENSE-2.0
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. default-domain:: cpp
+
+.. _alg_subgraph_isomorphism:
+
+====================
+Subgraph isomorphism
+====================
+
+.. include::  ../../../includes/graph/subgraph-isomorphism-introduction.rst
+
+------------------------
+Mathematical formulation
+------------------------
+
+.. _subgraph_isomorphism_compute:
+
+Subgraphs definition
+--------------------
+
+A graph :math:`H = (V'; E')` is called a subgraph of graph :math:`G = (V; E)` if 
+:math:`V' \subseteq V; E' \subseteq E` and :math:`V'` contains all the endpoints of all the 
+edges in :math:`E'` [Gross2014]_.
+
+Further we denote the :capterm:`induced subgraph on the vertex set<Induced subgraph on the vertex set>` 
+as **induced** subgraph, the :capterm:`induced subgraph on the edge set<Induced subgraph on the edge set>`
+as **non-induced** subgraph.
+
+Computing
+---------
+
+Given two graphs :math:`G` and :math:`H`, the problem is to define whether graph :math:`G` contains
+a subgraph which is isomorphic to graph :math:`H` and find exact mapping of subgraph :math:`H` in
+graph :math:`G`.
+
+Mapping is bijection or one-to-one correspondence between vertices of :math:`H` and subgraph of 
+graph :math:`G` with correspond vertex adjacensy relationship preserving for non-induced subgraph 
+and adjacensy and non-adjacensy preserving relationship for induced case. :math:`G` is called
+**target** graph, :math:`H` is called **pattern** graph.
+
+
+.. figure:: images/subgraph-isomorphism-target-pattern.png
+    :width: 600
+    :align: center
+    :alt:
+
+The only mapping for subgraph :math:`H` in graph :math:`G` from picture above in induced case is: 
+[3, 0, 1, 4, 2, 5]. Which means that:
+
+* pattern vertex with id 0 is mapped on vertex in target graph with id 3; 
+* pattern vertex with id 1 is mapped on vertex in target graph with id 0;
+* pattern vertex with id 2 is mapped on vertex in target graph with id 1;
+* pattern vertex with id 3 is mapped on vertex in target graph with id 4.
+* pattern vertex with id 4 is mapped on vertex in target graph with id 2.
+* pattern vertex with id 5 is mapped on vertex in target graph with id 5.
+
+
+Mappings for subgraph :math:`H` in graph :math:`G` in non-induced case are: [3, 0, 1, 4, 2, 5] and [3, 6, 1, 4, 2, 5].
```

Comment:
```suggestion
For the example above, the mappings for subgraph :math:`H` in graph :math:`G` are:

- Induced: [3, 0, 1, 4, 2, 5]
- Non-induced: [3, 0, 1, 4, 2, 5] and [3, 6, 1, 4, 2, 5]

The notation [3, 0, 1, 4, 2, 5] means that:

* pattern vertex with id 0 is mapped on vertex in target graph with id 3
* pattern vertex with id 1 is mapped on vertex in target graph with id 0
* pattern vertex with id 2 is mapped on vertex in target graph with id 1
* pattern vertex with id 3 is mapped on vertex in target graph with id 4
* pattern vertex with id 4 is mapped on vertex in target graph with id 2
* pattern vertex with id 5 is mapped on vertex in target graph with id 5
```

I'd suggest to rewrite the list that explains the mappings. For example, like this:

```
a vertex from the pattern graph with id=0 is mapped on a vertex in the target graph with id=3
```

or `onto a vertex`?

Reply:
both are ok